apps/aecore/src/aec_upnp.erl

```diff
@@ -0,0 +1,117 @@
+%%%=============================================================================
+%%% @copyright (C) 2019, Aeternity Anstalt
+%%% @doc
+%%%    Service handling UPnP and NAT-PMP protocols discovery and port mappings,
+%%%    to help to achieve better P2P connectivity. This works for routers with
+%%%    UPnP or NAT-PNP enabled, and should be used when there is no port
+%%%    forwarding configured.
+%%% @end
+%%%=============================================================================
+
+-module(aec_upnp).
+
+-behaviour(gen_server).
+
+%% API
+-export([is_enabled/0]).
+
+%% gen_server API
+-export([start_link/0]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+%% Port mapping lifetime in seconds.
+%% NAT-PMP RFC (https://tools.ietf.org/html/rfc6886) recommends to set it
+%% to 7200 seconds (two hours). No recommendation for UPnP found.
+-define(MAPPING_LIFETIME, 7200).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+is_enabled() ->
+    case aeu_env:find_config([<<"sync">>, <<"upnp_enabled">>],
+                             [user_config, schema_default]) of
+        {ok, IsEnabled} -> IsEnabled;
+        undefined       -> false
+    end.
+
+%%%===================================================================
+%%% Gen server API
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+init(Args) ->
+    epoch_sync:info("Starting UPnP/NAT-PMP service"),
+    erlang:send_after(rand:uniform(timer:seconds(1)), self(), add_port_mapping),
```

Comment:
```suggestion
    erlang:send_after(rand:uniform(1000), self(), add_port_mapping),
```
the API function timer:seconds(1) is horrible, the name does not convey what it does (gives the number of milliseconds for a number of seconds)...